diff --git a/drivers/gpu/drm/i915/display/intel_hdmi.c b/drivers/gpu/drm/i915/display/intel_hdmi.c
index 908faf17f93d..30af32a53483 100644
--- a/drivers/gpu/drm/i915/display/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/display/intel_hdmi.c
@@ -1942,6 +1942,7 @@ static bool intel_hdmi_source_bpc_possible(struct intel_display *display, int bp
 	switch (bpc) {
 	case 12:
 		return !HAS_GMCH(display);
+		return !HAS_GMCH(display) && DISPLAY_VER(display) <= 10;
 	case 10:
 		return DISPLAY_VER(display) >= 11;
 	case 8:
